### ISSUE TITLE
com.homebridge.server.plist issue on macOS Catalina

### ISSUE THREAD

**Issue by tannebil** [oznu/homebridge-config-ui-x#805]
**Describe The Bug:**
Using macOS Catalina com.homebridge.server.plist doesn't work unless moved to ~/Library/LaunchAgents

**To Reproduce:**
Install homebridge using the instructions for macOS in the Wiki on a fresh macOS Catalina install
Install dummy-switch plug-in and configure a few switches
Add to Homekit, verify switches
Reboot and login
dummy switches show as not responding in Home

**Expected behavior:**

dummy switches should show as active

Problem can fixed with either a "sudo hb-service restart" after every reboot or by moving com.homebridge.server.plist from /Library/LaunchDaemons to ~/Library/LaunchAgents

**Logs:**

log using restart

`
[8/8/2020, 3:12:00 AM] [HB Supervisor] Homebridge Storage Path: /Users/btanner/.homebridge
[8/8/2020, 3:12:00 AM] [HB Supervisor] Homebridge Config Path: /Users/btanner/.homebridge/config.json
[8/8/2020, 3:12:00 AM] [HB Supervisor] Logging to /Users/btanner/.homebridge/homebridge.log
[8/8/2020, 3:12:00 AM] [HB Supervisor] OS: Darwin 19.6.0 x64
[8/8/2020, 3:12:00 AM] [HB Supervisor] Node.js v12.16.2 /usr/local/bin/node
[8/8/2020, 3:12:00 AM] [HB Supervisor] Homebridge Path: /usr/local/lib/node_modules/homebridge/bin/homebridge
[8/8/2020, 3:12:00 AM] [HB Supervisor] UI Path: /usr/local/lib/node_modules/homebridge-config-ui-x/dist/bin/standalone.js
[8/8/2020, 3:12:00 AM] [HB Supervisor] Starting Homebridge with extra flags: -I
[8/8/2020, 3:12:00 AM] [HB Supervisor] Started Homebridge v1.1.1 with PID: 262
[8/8/2020, 3:12:01 AM] Loaded config.json with 7 accessories and 1 platforms.
[8/8/2020, 3:12:01 AM] ---
[8/8/2020, 3:12:01 AM] Loaded plugin: homebridge-config-ui-x@4.24.0
[8/8/2020, 3:12:01 AM] Registering platform 'homebridge-config-ui-x.config'
[8/8/2020, 3:12:01 AM] ---
[8/8/2020, 3:12:01 AM] Loaded plugin: homebridge-dummy@0.4.1
[8/8/2020, 3:12:01 AM] Registering accessory 'homebridge-dummy.DummySwitch'
[8/8/2020, 3:12:01 AM] ---
[8/8/2020, 3:12:01 AM] Loading 1 platforms...
[8/8/2020, 3:12:01 AM] [Config] Initializing config platform...
[8/8/2020, 3:12:01 AM] [Config] Running in Service Mode
[8/8/2020, 3:12:01 AM] Loading 7 accessories...
[8/8/2020, 3:12:01 AM] [Bill Occupancy 1] Initializing DummySwitch accessory...
[8/8/2020, 3:12:01 AM] [Bill Occupancy 1] Setting switch to true
[8/8/2020, 3:12:01 AM] [Chris Occupancy 1] Initializing DummySwitch accessory...
[8/8/2020, 3:12:01 AM] [Chris Occupancy 1] Setting switch to true
[8/8/2020, 3:12:01 AM] [Joint Occupancy 1] Initializing DummySwitch accessory...
[8/8/2020, 3:12:01 AM] [Joint Occupancy 1] Setting switch to true
[8/8/2020, 3:12:01 AM] [Laundry Room Kittens] Initializing DummySwitch accessory...
[8/8/2020, 3:12:01 AM] [Laundry Room Kittens] Setting switch to true
[8/8/2020, 3:12:01 AM] [Patio Motion Timer] Initializing DummySwitch accessory...
[8/8/2020, 3:12:01 AM] [Garage Door Open Timer] Initializing DummySwitch accessory...
[8/8/2020, 3:12:01 AM] [Garage Door Open Timer] Setting switch to false
[8/8/2020, 3:12:01 AM] [Garage Door Left Open Timer] Initializing DummySwitch accessory...
[8/8/2020, 3:12:01 AM] [Garage Door Left Open Timer] Setting switch to false
Setup Payload:
X-HM://0024J1OBOH2PA
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
   ┌────────────┐     
    │ 640-17-508 │     
   └────────────┘     
                       
[8/8/2020, 3:12:01 AM] Homebridge is running on port 51764.
[8/8/2020, 3:12:02 AM] [Config] Homebridge Config UI X v4.24.0 is listening on :: port 8581
[8/8/2020, 3:25:36 AM] [HB Supervisor] Stopping services...
[8/8/2020, 3:25:36 AM] Got SIGTERM, shutting down Homebridge...
[8/8/2020, 3:25:43 AM] [HB Supervisor] Homebridge Storage Path: /Users/btanner/.homebridge
[8/8/2020, 3:25:43 AM] [HB Supervisor] Homebridge Config Path: /Users/btanner/.homebridge/config.json
[8/8/2020, 3:25:43 AM] [HB Supervisor] Logging to /Users/btanner/.homebridge/homebridge.log
[8/8/2020, 3:25:43 AM] [HB Supervisor] OS: Darwin 19.6.0 x64
[8/8/2020, 3:25:43 AM] [HB Supervisor] Node.js v12.16.2 /usr/local/bin/node
[8/8/2020, 3:25:43 AM] [HB Supervisor] Homebridge Path: /usr/local/lib/node_modules/homebridge/bin/homebridge
[8/8/2020, 3:25:43 AM] [HB Supervisor] UI Path: /usr/local/lib/node_modules/homebridge-config-ui-x/dist/bin/standalone.js
[8/8/2020, 3:25:43 AM] [HB Supervisor] Starting Homebridge with extra flags: -I
[8/8/2020, 3:25:43 AM] [HB Supervisor] Started Homebridge v1.1.1 with PID: 1115
[8/8/2020, 3:25:43 AM] Loaded config.json with 7 accessories and 1 platforms.
[8/8/2020, 3:25:43 AM] ---
[8/8/2020, 3:25:44 AM] Loaded plugin: homebridge-config-ui-x@4.24.0
[8/8/2020, 3:25:44 AM] Registering platform 'homebridge-config-ui-x.config'
[8/8/2020, 3:25:44 AM] ---
[8/8/2020, 3:25:44 AM] Loaded plugin: homebridge-dummy@0.4.1
[8/8/2020, 3:25:44 AM] Registering accessory 'homebridge-dummy.DummySwitch'
[8/8/2020, 3:25:44 AM] ---
[8/8/2020, 3:25:44 AM] Loading 1 platforms...
[8/8/2020, 3:25:44 AM] [Config] Initializing config platform...
[8/8/2020, 3:25:44 AM] [Config] Running in Service Mode
[8/8/2020, 3:25:44 AM] Loading 7 accessories...
[8/8/2020, 3:25:44 AM] [Bill Occupancy 1] Initializing DummySwitch accessory...
[8/8/2020, 3:25:44 AM] [Bill Occupancy 1] Setting switch to true
[8/8/2020, 3:25:44 AM] [Chris Occupancy 1] Initializing DummySwitch accessory...
[8/8/2020, 3:25:44 AM] [Chris Occupancy 1] Setting switch to true
[8/8/2020, 3:25:44 AM] [Joint Occupancy 1] Initializing DummySwitch accessory...
[8/8/2020, 3:25:44 AM] [Joint Occupancy 1] Setting switch to true
[8/8/2020, 3:25:44 AM] [Laundry Room Kittens] Initializing DummySwitch accessory...
[8/8/2020, 3:25:44 AM] [Laundry Room Kittens] Setting switch to true
[8/8/2020, 3:25:44 AM] [Patio Motion Timer] Initializing DummySwitch accessory...
[8/8/2020, 3:25:44 AM] [Garage Door Open Timer] Initializing DummySwitch accessory...
[8/8/2020, 3:25:44 AM] [Garage Door Open Timer] Setting switch to false
[8/8/2020, 3:25:44 AM] [Garage Door Left Open Timer] Initializing DummySwitch accessory...
[8/8/2020, 3:25:44 AM] [Garage Door Left Open Timer] Setting switch to false
Setup Payload:
X-HM://0024J1OBOH2PA
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 640-17-508 │     
    └────────────┘     
                       
[8/8/2020, 3:25:44 AM] Homebridge is running on port 51764.
[8/8/2020, 3:25:44 AM] [Config] Homebridge Config UI X v4.24.0 is listening on :: port 8581
`

Log using LaunchAgents

`[8/8/2020, 9:24:47 PM] [HB Supervisor] Homebridge Storage Path: /Users/btanner/.homebridge
[8/8/2020, 9:24:47 PM] [HB Supervisor] Homebridge Config Path: /Users/btanner/.homebridge/config.json
[8/8/2020, 9:24:47 PM] [HB Supervisor] Logging to /Users/btanner/.homebridge/homebridge.log
[8/8/2020, 9:24:47 PM] [HB Supervisor] OS: Darwin 19.6.0 x64
[8/8/2020, 9:24:47 PM] [HB Supervisor] Node.js v12.16.2 /usr/local/bin/node
[8/8/2020, 9:24:47 PM] [HB Supervisor] Homebridge Path: /usr/local/lib/node_modules/homebridge/bin/homebridge
[8/8/2020, 9:24:47 PM] [HB Supervisor] UI Path: /usr/local/lib/node_modules/homebridge-config-ui-x/dist/bin/standalone.js
[8/8/2020, 9:24:47 PM] [HB Supervisor] Starting Homebridge with extra flags: -I
[8/8/2020, 9:24:47 PM] [HB Supervisor] Started Homebridge v1.1.1 with PID: 490
[8/8/2020, 9:24:48 PM] Loaded config.json with 7 accessories and 1 platforms.
[8/8/2020, 9:24:48 PM] ---
[8/8/2020, 9:24:48 PM] Loaded plugin: homebridge-config-ui-x@4.24.0
[8/8/2020, 9:24:48 PM] Registering platform 'homebridge-config-ui-x.config'
[8/8/2020, 9:24:48 PM] ---
[8/8/2020, 9:24:48 PM] Loaded plugin: homebridge-dummy@0.4.1
[8/8/2020, 9:24:48 PM] Registering accessory 'homebridge-dummy.DummySwitch'
[8/8/2020, 9:24:48 PM] ---
[8/8/2020, 9:24:48 PM] Loading 1 platforms...
[8/8/2020, 9:24:48 PM] [Config] Initializing config platform...
[8/8/2020, 9:24:48 PM] [Config] Running in Service Mode
[8/8/2020, 9:24:48 PM] Loading 7 accessories...
[8/8/2020, 9:24:48 PM] [Bill Occupancy 1] Initializing DummySwitch accessory...
[8/8/2020, 9:24:48 PM] [Bill Occupancy 1] Setting switch to true
[8/8/2020, 9:24:48 PM] [Chris Occupancy 1] Initializing DummySwitch accessory...
[8/8/2020, 9:24:48 PM] [Chris Occupancy 1] Setting switch to true
[8/8/2020, 9:24:48 PM] [Joint Occupancy 1] Initializing DummySwitch accessory...
[8/8/2020, 9:24:48 PM] [Joint Occupancy 1] Setting switch to true
[8/8/2020, 9:24:48 PM] [Laundry Room Kittens] Initializing DummySwitch accessory...
[8/8/2020, 9:24:48 PM] [Laundry Room Kittens] Setting switch to false
[8/8/2020, 9:24:48 PM] [Patio Motion Timer] Initializing DummySwitch accessory...
[8/8/2020, 9:24:48 PM] [Garage Door Open Timer] Initializing DummySwitch accessory...
[8/8/2020, 9:24:48 PM] [Garage Door Open Timer] Setting switch to false
[8/8/2020, 9:24:48 PM] [Garage Door Left Open Timer] Initializing DummySwitch accessory...
[8/8/2020, 9:24:48 PM] [Garage Door Left Open Timer] Setting switch to false
Setup Payload:
X-HM://0024J1OBOH2PA
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 640-17-508 │     
    └────────────┘     
                       
[8/8/2020, 9:24:48 PM] Homebridge is running on port 51764.
[8/8/2020, 9:24:49 PM] [Config] Homebridge Config UI X v4.24.0 is listening on :: port 8581
`

Show the Homebridge / Homebridge Config UI X logs here.
```

**Homebridge Config:** (user name and pin edited out)

`{
    "bridge": {
        "name": "Homebridge A6FE",
        "username": "",
        "port": 51764,
        "pin": ""
    },
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        }
    ],
    "accessories": [
        {
            "name": "Bill Occupancy 1",
            "stateful": true,
            "reverse": false,
            "time": 1000,
            "accessory": "DummySwitch"
        },
        {
            "name": "Chris Occupancy 1",
            "stateful": true,
            "reverse": false,
            "time": 1000,
            "accessory": "DummySwitch"
        },
        {
            "name": "Joint Occupancy 1",
            "stateful": true,
            "reverse": false,
            "time": 1000,
            "accessory": "DummySwitch"
        },
        {
            "name": "Laundry Room Kittens",
            "stateful": true,
            "reverse": false,
            "time": 600000,
            "accessory": "DummySwitch"
        },
        {
            "name": "Patio Motion Timer",
            "stateful": false,
            "reverse": false,
            "time": 360000,
            "accessory": "DummySwitch"
        },
        {
            "name": "Garage Door Open Timer",
            "stateful": true,
            "reverse": false,
            "time": 1000,
            "accessory": "DummySwitch"
        },
        {
            "name": "Garage Door Left Open Timer",
            "stateful": true,
            "reverse": false,
            "time": 1000,
            "accessory": "DummySwitch"
        }
    ]
}
`
```json
Show your homebridge config.json here
```

**Screenshots:**

**Environment:**

* **Node.js Version**:  
v12.16.2
* **NPM Version**:  
6.14.4
* **Homebridge Version**:  
1.1.1
* **Homebridge Config UI X Version**:
* **Operating System**: Raspbian / Ubuntu / Debian / Windows / macOS / Docker
macOS 10.15.6
* **Process Supervisor**: Docker / Systemd / init.d / pm2 / launchctl / hb-service / other / none
hb-service

**Comment by tannebil** [comment]
Moving com.homebridge.server.plist from /Library/LaunchDaemons to ~/Library/LaunchAgents seems to work around the problem.

**Comment by tannebil** [comment]
Built another new homebridge VM. Same problem. Needed to move the plist, do an uninstall/install using sudo hb-service, and make the account autologin for the VM.

Is Homebridge supposed to be able to run without an account logged in and just managed through a remote web session?

**Comment by oznu** [comment]
I've just tested this on an up-to-date macOS Catalina VM and can't replicate the issue. The service is starting as expected on reboot / restart **before** the user logs in as expected.

**Comment by tannebil** [comment]
I uninstalled/installed the hb-service on both and everything works exactly as you described. I guess I've run homebridge from a user account for so long that I got messed up not realizing it was now running it as a system process. It's obvious in retrospect (my launchd knowledge is thin) but I just kept charging ahead trying to fit it into my mental box.

Thanks for getting me unstuck. 

You are doing great work making homebridge a much easier program to use. One more VM left to move out of the dark ages of Mojave and running as a user program.
